pyanaconda/modules/boss/boss.py

```diff
@@ -35,10 +34,11 @@
 @dbus_interface(dbus_constants.DBUS_BOSS_NAME)
 class Boss(BaseModule):
 
-    def __init__(self):
+    def __init__(self, module_manager=None):
```

Comment:
I am not sure if we want to do this with every manager.

Reply:
Yep you want to. It is prevention from entangling the classes too much and be able to reuse the class for other purpose.

Also there shouldn't be more then a few managers. If there is more it should be redesigned a split otherwise you will get spaghetti at the end. Also it is helping to prevent Open Close Principle and SRP.

Reply:
What's the purpose of this ? Testing (eq. to more easily mock the manager class) ?

Reply:
The main reason is testing but also it is the correct way in general because you can replace parts used inside without changing the whole class. Look on the [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) if you want to find out more.